### PR TITLE
chore: upgrade sample apps to RN 0.69.4

### PR DESCRIPTION
**Pull request by vanGalilea**
## 🎯 Goal

Upgrades RN in example apps to 0.69.3 and React to 18.2.0

## 🛠 Implementation details

Following regular upgrade strategy via [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper) and upgrading other libs. that were blocking upgrades such as RN markdown

## 🎨 UI Changes

N/A

## 🧪 Testing

Run all example apps on all OS's

## ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] PR targets the `develop` branch
- [ ] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [ ] SampleApp iOS and Android
  - [ ] Expo iOS and Android

### CODE REVIEW

**Comment by madsroskar** [comment]
> and upgrading other libs. that were blocking upgrade such as

I think there's an end to this sentence that's missing? :)

File: package/expo-package/package.json [hunk 1]
```diff
@@ -17,7 +17,6 @@
     "expo": ">=44.0.0",
     "expo-document-picker": "^9.1.2",
     "expo-file-system": "^11.0.2",
-    "expo-haptics": "^10.0.0",
```
Comment (madsroskar) [inline package/expo-package/package.json:20]: Why is this removed? We use it in the (`expo-package`)[https://github.com/GetStream/stream-chat-react-native/blob/develop/package/expo-package/src/index.js#L8]

Comment (vanGalilea) [reply package/expo-package/package.json:20]: That's a mistake, thanks for spotting it

**Review by madsroskar — APPROVED** [review]
LGTM 👍🏻 Should we update the version compability docs as part of this, or do you want to do that separately?
